### ISSUE TITLE
Ubuntu TensorFlow-GPU 1.8.0 with CUDA 9.1, cuDNN 7.1, Python 3.6.5, AVX, AVX2, FMA, and SSE4.2

### ISSUE THREAD

**Issue by cedrickchee** [yaroslavvb/tensorflow-community-wheels#65]
Hi everyone,

I am aware that many of us are eagerly waiting for TensorFlow's team to release binaries that support CUDA 9.1. I have built a TensorFlow 1.8.0 wheel with the following support:
- GPU
- CUDA 9.1
- cuDNN 7.1.3
- Python 3.6.5
- CPU support: AVX, AVX2, FMA, and SSE4.2
- Optimized for AWS p2 instances running Ubuntu 16.04 LTS.

Build specs:

| Repo Path | Compiler | CUDA/cuDNN | SIMD | Compute Capability | Notes |
|-|-|-|-|-|-|
| 1.8.0/py36/GPU/cuda91_cudnn71_avx2_fma_sse4 | GCC 5.4.0 | 9.1.85.3/7.1.3 | x86_64 | 3.7 | Python 3.6.5. Compiler optimization flags: -march=native. jemalloc enabled. No XLA, OpenCL, MPI support. NCCL 1.3.4. CXX flag: D_GLIBCXX_USE_CXX11_ABI=0 |

Install using `pip`:

```bash
pip install https://github.com/cedrickchee/tensorflow-community-builds/releases/download/1.8.0_py36_gpu_cuda91_cudnn71_avx2_fma_sse4/tensorflow-1.8.0-cp36-cp36m-linux_x86_64.whl
```
More info is in [my repo](https://github.com/cedrickchee/tensorflow-community-builds). Enjoy :slightly_smiling_face:

**Comment by cipri-tom** [comment]
Sorry if I'm a noob, but you have to install CUDA 9.1 separately, right ? So far I've been using some docker images, like the official tensorflow ones, which have CUDA 9.0. Do you know of any images with 9.1 ?

**Comment by cedrickchee** [comment]
Yes, you are right on the separate installation of CUDA 9.1. I don't use Docker for my use case. So, I am not so aware of Docker images with 9.1.

**Comment by cipri-tom** [comment]
ok. Thank you!

And thank you for the wheel !

**Comment by MuTaTeD** [comment]
Hi,
It appears that cudnn 7.1.3 is no longer available on the nVidia site for cuda 9.1. only thing available is 7.1.4 for 9.2. Can you update and provide new wheel or maybe link to cudnn 7.1.3...

Thanks

**Comment by cedrickchee** [comment]
Link to cuDNN 7.1.3 for CUDA 9.1: http://developer.download.nvidia.com/compute/redist/cudnn/v7.1.3/cudnn-9.1-linux-x64-v7.1.tgz

**Comment by precabal** [comment]
newbie here: will this wheel work if I don't have cuda and want the wheel just for the cpu optimizations?

**Comment by cipri-tom** [comment]
@precabal I think it will, as long as you don't use hard device assignments i.e. `tf.device("/device:GPU:0")`. Try it and let us know 😃

**Comment by precabal** [comment]
thanks @cipri-tom, I installed it successfully but when trying to run an example I'm getting an error saying `ImportError: libcublas.so.9.1: cannot open shared object file: No such file or directory`. This happens when running `import tensorflow as tf`. After a naïve search in google it seems to me that it is a cuda related library that is not loading, so I'm stuck there... any help?

**Comment by cipri-tom** [comment]
ah... so then I was wrong and you cannot use it without valid CUDA installed :( . Sorry. I thought TF only loads CUDA when needed

**Comment by precabal** [comment]
no problem. it was worth trying it out. thanks for your reply!
